File: test/problems/2-medium/49-Group-Anagrams/solution.java
import java.util.*;

public List<List<String>> groupAnagrams(String[] strs) {
    List<List<String>> result = new ArrayList<>();
    Map<String, List<String>> groups = new HashMap<>();
    
    for (String s : strs) {
        String signature = getSignature(s);
        groups.computeIfAbsent(signature, k -> new ArrayList<>()).add(s);
    }
    
    for (List<String> group : groups.values()) {
        result.add(group);
    }
    
    return result;
}

private String getSignature(String s) {
    int[] count = new int[26];
    for (char c : s.toCharArray()) {
        count[c - 'a']++;
    }
    
    StringBuilder sb = new StringBuilder();
    for (int i = 0; i < 26; i++) {
        if (count[i] != 0) {
            sb.append((char)('a' + i)).append(count[i]);
        }
    }
    return sb.toString();
}

File: test/problems/2-medium/49-Group-Anagrams/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <sstream> 

using namespace std;

string getSignature(const string& s) {
    vector<int> count(26, 0);
    for (char c : s) {
        count[c - 'a']++;
    }

    stringstream ss;
    for (int i = 0; i < 26; i++) {
        if (count[i] != 0) {
            ss << (char)('a' + i) << count[i];
        }
    }
    return ss.str();
}

vector<vector<string>> groupAnagrams(vector<string>& strs) {
    vector<vector<string>> result;
    unordered_map<string, vector<string>> groups;

    for (const string& s : strs) {
        groups[getSignature(s)].push_back(s);
    }

    for (const auto& entry : groups) {
        result.push_back(entry.second);
    }

    return result;
}